xbmc/Util.cpp

```diff
@@ -1251,6 +1251,21 @@ int CUtil::GetMatchingSource(const std::string& strPath1,
   if (checkURL.IsProtocol("multipath"))
     strPath = CMultiPathDirectory::GetFirstPath(strPath);
 
+  // Check to see if protocol is VFS addon (eg. archive://)
+  if (!checkURL.GetProtocol().empty() && CServiceBroker::IsAddonInterfaceUp())
+  {
+    for (const auto& vfsAddon : CServiceBroker::GetVFSAddonCache().GetAddonInstances())
+    {
+      auto prots = StringUtils::Split(vfsAddon->GetProtocols(), "|");
+      if (vfsAddon->HasDirectories() &&
+          std::ranges::find(prots, checkURL.GetProtocol()) != prots.end())
+      {
+        strPath = checkURL.GetHostName(); // resolve host (eg. actual .zip file)
```

Comment:
I prefer short lines to help developers with large fonts or small screens. We don't pay by the line, so something like:

```c++

      {
        // Resolve host (eg. actual .zip file)
        strPath = checkURL.GetHostName();
        break;
      }
```

Also, as you saw, comments being sentences, I like Sentence casing too for code clarity.